android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/webextension/GeckoWebExtension.kt

```diff
@@ -351,11 +356,28 @@ class GeckoWebExtension(
         return nativeExtension.metaData.let {
             Metadata(
                 name = it.name,
+                fullDescription = it.fullDescription,
+                downloadUrl = it.downloadUrl,
+                updateDate = it.updateDate,
+                averageRating = it.averageRating.toFloat(),
+                reviewCount = it.reviewCount,
                 description = it.description,
                 developerName = it.creatorName,
                 developerUrl = it.creatorUrl,
                 homePageUrl = it.homepageUrl,
+                creatorName = it.creatorName,
+                creatorUrl = it.creatorUrl.orEmpty(),
+                reviewUrl = it.reviewUrl,
                 version = it.version,
+                getIcon = {
+                    try {
+                        withTimeoutOrNull(EXTENSION_ICON_RETRIEVE_TIMEOUT) {
+                            it.icon.getBitmap(EXTENSION_ICON_IMAGE_SIZE).await()
```

Comment:
is this blocking?

Reply:
Yes, between the contexts of the [suspended](https://kotlinlang.org/docs/composing-suspending-functions.html#concurrent-using-async) function. The consumer of the function will define in which thread function will run. We are using a [similar pattern for loading media session artwork from Gecko](https://github.com/mozilla-mobile/firefox-android/blob/638e0b89ed7e5267224c3fa45081daf37371d9ee/android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/mediasession/GeckoMediaSessionDelegate.kt#L40-L50).

Reply:
Noticed we already had a previous solution for loading icon for temporary extension used the same approach.